fe/fe-core/src/main/java/com/starrocks/transaction/DatabaseTransactionMgr.java

```diff
@@ -690,7 +690,7 @@ public boolean canTxnFinished(TransactionState txn, Set<Long> errReplicas, Set<L
                             // if most replica's version have been updated to version published
                             // which means publish version task finished in replica  
                             for (Replica replica : ((LocalTablet) tablet).getReplicas()) {
-                                if (!errReplicas.contains(replica.getId()) && replica.getLastFailedVersion() < 0) {
```

Comment:
When `replica.getLastFailedVersion() < 0` is false, healthReplicaNum won't be added, and it won't be greater than quorumNum. And this function will return false.
Your changing will only make healthReplicaNum become greater, which can not resolve the problem in your PR description.

Reply:
it is the problem of errorReplica. I keep the condition getLastFailedVersion < 0 for healthyNum counter and remove it for errorReplicas.